### PR TITLE
feat(rbac): integrate rbac plugin with the pre-created database generated by Backstage using TypeORM adapter.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
# Referenced issue:

fixes:  janus-idp/backstage-showcase#359

# What does this pull request do:

Integrate rbac plugin with the pre-created database generated by Backstage using TypeORM adapter. Add support two databases: PostgreSQL for production and better-sqlite3 for development purposes.

# Demo:

https://youtu.be/27y4IuuQVkw

### CODE REVIEW

**Comment by tumido** [comment]
Hey @AndrienkoAleksandr (and @PatAKnight in the previous PRs), it would be easier if you followed conventional commits for PR titles/commit messages. The CI check would then pass and Release pipeline has a chance to pick up your changes for changelog generation and package version inference.

https://www.conventionalcommits.org/en/v1.0.0/

**Comment by AndrienkoAleksandr** [comment]
> Hey @AndrienkoAleksandr (and @PatAKnight in the previous PRs), it would be easier if you followed conventional commits for PR titles/commit messages

Ok, thanks. I fixed commit messages and checked them using validator: https://commitlint.io/ , but ci check any way is failing...

File: plugins/rh-rbac-backend/src/service/policy-builder.ts [hunk 1]
```diff
@@ -35,27 +35,20 @@ export class PolicyBuilder {
     discovery: PluginEndpointDiscovery;
     identity: IdentityApi;
     permissions: PermissionEvaluator;
+    database: PluginDatabaseManager;
   }): Promise<Router> {
-    // TODO: Replace with a DB adapter.
-    const fileAdapter = new FileAdapter(
```
Comment (PatAKnight) [inline plugins/rh-rbac-backend/src/service/policy-builder.ts:40]: Do we want to leave this in as there is an additional issue on making the [file adapter configurable](https://github.com/janus-idp/backstage-showcase/issues/360)?

**Comment by jasperchui** [comment]
https://github.com/janus-idp/backstage-showcase/issues/359

File: plugins/rh-rbac-backend/src/service/policy-builder.ts (hunk 1 above)
Comment (AndrienkoAleksandr) [reply plugins/rh-rbac-backend/src/service/policy-builder.ts:40]: Ok, I turned back this code.

File: plugins/rh-rbac-backend/config.d.ts [hunk 2]
```diff
@@ -18,5 +18,16 @@ export interface Config {
         }>;
       };
     };
+    /*
+     * Optional configuration for database
+     * @visibility frontend
+     */
+    database?: {
```
Comment (gorkem) [inline plugins/rh-rbac-backend/config.d.ts:25]: this should be under rbac

File: plugins/rh-rbac-backend/src/service/casbin-adapter-factory.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { ConfigApi } from '@backstage/core-plugin-api';
+
+import { Adapter } from 'casbin';
+import TypeORMAdapter from 'typeorm-adapter';
+
+import { resolve } from 'path';
+
+const DEFAULT_SQLITE3_STORAGE_FILE_NAME = 'rh-rbac-backend.sqlite';
+
+export class CasbinAdapterFactory {
```
Comment (gorkem) [inline plugins/rh-rbac-backend/src/service/casbin-adapter-factory.ts:11]: This only creates DB adaptors rename it to `CasbinDBAdaptorFactory`

**Review by gorkem — COMMENTED** [review]
minor naming changes and we are good to go

**Review by gorkem — APPROVED** [review]
lgtm

**Pull request merged**